pkg/planner/core/exhaust_physical_plans.go

```diff
@@ -1535,7 +1535,16 @@ func constructIndexJoinInnerSideTaskWithAggCheck(p *logicalop.LogicalJoin, prop
 
 	// build hash agg, when the stream agg is illegal such as the order by prop is not matched
 	if aggTask == nil {
-		physicalHashAgg := NewPhysicalHashAgg(la, la.StatsInfo(), prop)
+		stats := la.StatsInfo()
```

Comment:
Could we build this stats before build the physicalxxxAgg ? Then we can use this new stats in the streamagg and the hashagg. Don't compute the stats twice

Reply:
Actually, the change in his stats is also due to changes in the planner. Therefore, recalculation is needed when changes occur. Currently, the stats for all NewPhysicalHashAgg instances are computed in real time when they are called.

Reply:
@elsa0520, the stats from indexPlan or tablePlan is recomputed for single time, the original logic stats is derived for entire inner side. But i think we can merge this two logic together because we don't care it's is stream or hash one.

Reply:
I have merged the two parts of the code.